packages/compass-aggregations/src/components/pipeline-explain/index.tsx

```diff
@@ -1,36 +1,112 @@
-import React from 'react';
-import { Modal, H3, Body } from '@mongodb-js/compass-components';
+import React, { useEffect } from 'react';
+import {
+  css,
+  spacing,
+  Modal,
+  CancelLoader,
+  H3,
+  ModalFooter,
+  Button,
+  ErrorSummary,
+} from '@mongodb-js/compass-components';
 import { connect } from 'react-redux';
 
 import type { RootState } from '../../modules';
-import { closeExplainModal } from '../../modules/explain';
+import type { ExplainData } from '../../modules/explain';
+import {
+  explainAggregation,
+  closeExplainModal,
+  cancelExplain,
+} from '../../modules/explain';
+import { ExplainResults } from './explain-results';
 
 type PipelineExplainProps = {
   isModalOpen: boolean;
+  isLoading: boolean;
+  error?: string;
+  explain?: ExplainData;
   onCloseModal: () => void;
+  onRunExplain: () => void;
+  onCancelExplain: () => void;
 };
 
+const contentStyles = css({
+  marginTop: spacing[3],
+  marginBottom: spacing[3],
+});
+
+const footerStyles = css({
+  paddingRight: 0,
+  paddingBottom: 0,
+});
+
 export const PipelineExplain: React.FunctionComponent<PipelineExplainProps> = ({
   isModalOpen,
+  isLoading,
+  error,
+  explain,
   onCloseModal,
+  onRunExplain,
+  onCancelExplain,
 }) => {
+  useEffect(() => {
+    isModalOpen && onRunExplain();
+  }, [isModalOpen, onRunExplain]);
+
+  let content = null;
+  if (isLoading) {
+    content = (
+      <CancelLoader
+        dataTestId="pipeline-explain-cancel"
+        cancelText="Cancel"
+        onCancel={() => onCancelExplain()}
+        progressText="Running explain"
+      />
+    );
+  } else if (error) {
+    content = (
+      <ErrorSummary data-testid="pipeline-explain-error" errors={[error]} />
+    );
+  } else if (explain) {
+    content = <ExplainResults plan={explain.plan} stats={explain.stats} />;
+  }
+
+  const modalFooter = (
+    <ModalFooter className={footerStyles}>
+      <Button
+        onClick={onCloseModal}
+        data-testid="pipeline-explain-footer-close-button"
+      >
+        Close
+      </Button>
+    </ModalFooter>
+  );
```

Comment:
I would suggest you to do these types of things only for the parts of the rendering tree that are complex to grasp at a glance or have a non-trivial branching logic. Otherwise this has the same problem that having these `renderThing` methods that we have all over the place: you can't get a view of a component structure at a glance and have to jump back and forth to understand it